src/scratchOrgInfoApi.ts

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Node
+import * as fs from 'fs';
+
+// @salesforce
+import { Optional } from '@salesforce/ts-types';
+import { env, Duration, upperFirst } from '@salesforce/kit';
+import { ensureString, getString } from '@salesforce/ts-types';
+
+// Thirdparty
+import { RecordResult, OAuth2Options } from 'jsforce';
+import { retry, retryDecorator } from 'ts-retry-promise';
+
+// Local
+import { Org } from './org';
+import { Logger } from './logger';
+import mapKeys from './util/mapKeys';
+import { AuthInfo } from './authInfo';
+import { Messages } from './messages';
+import { SfdxError } from './sfdxError';
+import { SfdcUrl } from './util/sfdcUrl';
+import { MyDomainResolver } from './status/myDomainResolver';
+
+import SettingsGenerator, { ScratchDefinition } from './scratchOrgSettingsGenerator';
+
+export interface ScratchOrgInfo {
```

Comment:
There's really 2 flavors of this type: a request (some things won't exist, others are required, some can't be set) and what exists on the server after creation (more things are guaranteed to exist, all are readonly) that's in the response.

Think of the other code that's using it (org:display|list, org:delete) that could benefit from having pretty fantastic typings in sfdx-core.